### ISSUE TITLE
Weights for vector with missing values

### ISSUE THREAD

**Issue by jariji** [JuliaStats/StatsBase.jl#825]
I want to convert a numeric vector with missings to weights but it doesn't work.

```jl
julia> pweights([1,2,3,missing])
ERROR: MethodError: no method matching pweights(::Vector{Union{Missing, Int64}})
```

**Comment by ararslan** [comment]
How would you want the missing values to be treated? That is, would you expect the corresponding weight value to be `missing` or would you expect that missing values would be skipped when calling `pweights`? If the latter, `pweights(skipmissing(...))` should work I think.

**Comment by jariji** [comment]
I meant for the weight to be `missing`.

**Comment by ararslan** [comment]
Okay. It wouldn't be unreasonable to allow that but I'm not sure it would be that useful. The various `AbstractWeights` subtypes precompute and store the sum of the weights but if one or more of the provided weight values is `missing`, the sum will be `missing`, which means that all weighted statistics, variance corrections, etc. will be `missing`.

**Comment by nalimilan** [comment]
Could you describe your use case? As @ararslan said `missing` weights don't seem very useful as all operations are going to return `missing` (or fail). Actually, we're rather willing to be stricter and disallow passing `NaN` and `Inf` values for weights, see https://github.com/JuliaStats/StatsBase.jl/pull/814.

**Comment by jariji** [comment]
I have a dataframe with a weights column, some of which are missing. I convert the weights from float to `ProbabilityWeights` as part of my initial cleaning. Then when I'm ready to run a regression, I drop the missing values and fit the GLM with the weights. 

Currently I have to postpone converting the weights until after I've dropped the missing values. I didn't expect to have to do that, but if it doesn't make sense to have missing weights that's ok.

**Comment by bkamins** [comment]
> I didn't expect to have to do that, but if it doesn't make sense to have missing weights that's ok.

The problem is that, by design, `Weights` carries over its `sum` internally for performance and having `missing` values invalidates this design decision.

**Comment by jariji** [comment]
Is that the case for `ProbabilityWeights` too?

**Comment by bkamins** [comment]
Yes - the structure is generated with `@weights` macro just like all other weights:
https://github.com/JuliaStats/StatsBase.jl/blob/bd4ca61f4bb75f2c6cd0a47aee1cfde7b696eb9c/src/weights.jl#L173

**Comment by nalimilan** [comment]
That said, even if computing and storing the sum internally isn't useful in the presence of missing values, it should work fine, so this could be allowed. What worries me more is the increased complexity generated by allowing `<: Union{Real, Missing}` eltype and not only `<: Real`.

If you need to drop observations with missing weights anyway, then I'd say it's fine to require you to do that before constructing the `ProbabilityWeights` object. In general it seems hard to do anything when the weight is missing anyway.
